### DISCUSSION TITLE
v1.9.1 Better colors 🎉

### DISCUSSION THREAD

**Discussion by luisbocanegra** [luisbocanegra/kde-material-you-colors#205] [Announcements]
This release switches to the updated Material You python library by @T-Dynamos
New scheme variants, colorfulness/brightness control, and more:

## New distribution packages
- openSUSE Tumbleweed [install instructions](https://software.opensuse.org//download.html?project=home%3Aluisbocanegra&package=kde-material-you-colors)
- Fedora 40 and Fedora Rawhide [install instructions](https://copr.fedorainfracloud.org/coprs/luisbocanegra/kde-material-you-colors/)

Arch Linux AUR and pipx install/update instructions stay the same, check [README](https://github.com/luisbocanegra/kde-material-you-colors?tab=readme-ov-file#installing)
### Bug fixes

* Respect screenshot delay https://github.com/luisbocanegra/kde-material-you-colors/commit/4caf11ae543a92cdec15c14428722dfc1122cfd8
* Don't apply delay to non-screenshot method https://github.com/luisbocanegra/kde-material-you-colors/commit/974e318a2405ee8d5afe668690e0193b21359e0d
* Fix widget not loading backend colors after username changes https://github.com/luisbocanegra/kde-material-you-colors/commit/05d60bc059d7a7dc8a396a62eb4f0dec8da61da8
* Fix autostart and launch desktop entries https://github.com/luisbocanegra/kde-material-you-colors/commit/c2a9544848be04f93175ea435c596d1bf9374c4b
* Allow custom qdbus executable name https://github.com/luisbocanegra/kde-material-you-colors/commit/e06de47756fed48fcc171ef388826592b8c11a5f

### Improvements

* Port to T-Dynamos/materialyoucolor-python in https://github.com/luisbocanegra/kde-material-you-colors/pull/192
* Add option to change between Material You scheme variants https://github.com/luisbocanegra/kde-material-you-colors/commit/6f47f166514787caded0ae6db5b42d89e338dfd6
* Better Plasma color scheme https://github.com/luisbocanegra/kde-material-you-colors/commit/60cfbd0fd14c0bf082ffd04d4d114767107f256e
* Stronger Plasma color scheme Disabled effect https://github.com/luisbocanegra/kde-material-you-colors/commit/b74339d9eaeea39c77432a68188d3bb271aa2acd
* Simplify generated theme https://github.com/luisbocanegra/kde-material-you-colors/commit/c22f8ae94ff003924856a84dc12472893041b71e
* Use dynamic color for custom text roles in https://github.com/luisbocanegra/kde-material-you-colors/pull/200
* Add option to change colorfulness and brightness of theme (Replace blend with colorfulness and brightness) in https://github.com/luisbocanegra/kde-material-you-colors/pull/201

### Other

* Default to the top color when the requested index is out of bounds https://github.com/luisbocanegra/kde-material-you-colors/commit/b9338a14f7b5a5dad2d6061671a532e6aba0fd31
* plasmoid: Add links cursor feedback https://github.com/luisbocanegra/kde-material-you-colors/commit/cc434a4d114824bd90c079aa9e803deba52917d4
* Add back Pillow dependency https://github.com/luisbocanegra/kde-material-you-colors/commit/0a0fd50ce04acd56bfe5ada5cb321457674bdc7f
* Require QT6/KF6 to build https://github.com/luisbocanegra/kde-material-you-colors/commit/914f7d7afbd93b1586ec6b5b94887e2097f9d319

**Note: The generated theme json structure has changed a, if you make use of it to theme other things, you will need to update those scripts**

**Full Changelog**: https://github.com/luisbocanegra/kde-material-you-colors/compare/v1.8.0...v1.9.1

![191-1](https://github.com/luisbocanegra/kde-material-you-colors/assets/15076387/09d4bce3-6dd4-468b-915e-e72583019c2a)

![screenshot1](https://github.com/luisbocanegra/kde-material-you-colors/assets/15076387/7a32c86f-2c8c-4f44-96dd-d1a85c88993c)

![191-2](https://github.com/luisbocanegra/kde-material-you-colors/assets/15076387/f215f783-0fbb-4906-a009-93ede7d6bb56)

<hr /><em>This discussion was created from the release <a href='https://github.com/luisbocanegra/kde-material-you-colors/releases/tag/v1.9.1'>v1.9.1 Better colors 🎉</a>.</em>

**Comment by ShatteredSkies** [comment]
darker title bar seems to be enabled on all windows, i can't figure out how to disable it.

`darker_window_list` is left empty

![qwefwefwefe](https://github.com/luisbocanegra/kde-material-you-colors/assets/56289692/d40748cb-7b92-4bdd-9146-4411674f6392)

**Reply by ShatteredSkies** [reply]
okay i just looked at your screenshot and figured it's actually the tool bar that has the wrong color.

**Reply by luisbocanegra** [reply]
Yeah, Lightly theme applies a different color to them and other things. If you open a QWidgets application (Dolphin,kate...) you'll notice the toolbar is inconsistent compared to applications using Kirigami (Spectacle, System Settings...).

Whereas with the Breeze theme all apps have a consistent toolbar color. In Plasma 5 I prioritized Lightly over Breeze but now I only use Breeze and this is reflected in those changes.

Since the view, toolbar and title bar have the same color now, the `darker_window_list` it's redundant. Maybe I could convert it into "brighter titlebar to match Lightly" instead?

Ultimately, those problems could be resolved by making the color schemes fully customizable with templates, like pywal does, but we'll see.

**Reply by ShatteredSkies** [reply]
that's unfortunate, default breeze theme appears too inconsistent for me to use.

"brighter titlebar to match Lightly" would make things look too monotonous imo.
don't know how possible it is but i rather have a "darker toolbar" option.

**Reply by luisbocanegra** [reply]
In Lightly toolbar color is "connected" to the window color:

https://github.com/luisbocanegra/kde-material-you-colors/blob/0f89fc8f61d30b4d8c5279f167fc7e69f3eca6cc/src/kde_material_you_colors/schemeconfigs.py#L502

So trying to match titlebar+toolbar but have a different window (places sidebar) color is not possible with Lightly:

![Screenshot_20240521_134356](https://github.com/luisbocanegra/kde-material-you-colors/assets/15076387/fc3b2440-b00c-4e4e-b311-1774c678ecec)

In 1.8.0 I applied the same brighter color for window+header, and a darker one for the view so the darker titlebar made sense. Maybe we could go back to that but using [surface container roles](https://m3.material.io/styles/color/roles)

**Reply by ShatteredSkies** [reply]
oh well.. that's a weird choice from lightly.

surface container roles might work, just need to see how it looks.
